pygfx/utils/cm.py

```diff
@@ -22,14 +35,110 @@
     _cividis_data,
 )
 
-from ..resources import Texture, TextureMap
+
+_data = {
+    # Simple maps
+    "gray": [(0, 0, 0, 0), (1, 1, 1, 1)],
+    "cool": [(0, 1, 1), (1, 0, 1)],
+    "hot": [(0, 0, 0), (1, 0, 0), (1, 1, 0), (1, 1, 1)],
+    "bone": [(0, 0, 0), (0.333, 0.333, 0.444), (0.666, 0.777, 0.777), (1, 1, 1)],
+    "copper": [(0, 0, 0), (1, 0.7, 0.5)],
+    "pink": [(0.1, 0, 0), (0.75, 0.5, 0.5), (0.9, 0.9, 0.7), (1, 1, 1)],
+    "spring": [(1, 0, 1), (1, 1, 0)],
+    "summer": [(0, 0.5, 0.4), (1, 1, 0.4)],
+    "autumn": [(1, 0, 0), (1, 1, 0)],
+    "winter": [(0, 0, 1), (0, 1, 0.5)],
+    "jet": [
+        (0, 0, 0.5),
+        (0, 0, 1),
+        (0, 0.5, 1),
+        (0, 1, 1),
+        (0.5, 1, 0.5),
+        (1, 1, 0),
+        (1, 0.5, 0),
+        (1, 0, 0),
+        (0.5, 0, 0),
+    ],
+    # Larger maps
+    "viridis": _viridis_data,
+    "plasma": _plasma_data,
+    "inferno": _inferno_data,
+    "magma": _magma_data,
+    "cividis": _cividis_data,
+}
+
+__all__ = sorted([*_data.keys(), "create_colormap"])
+
+
+def create_colormap(input, n=256):
+    """Create a colormap from given data.
+
+    Returns a TextureMap with (at least) n colors.
+
+    If a list or array-like object is given, the values are interpolated to create the map.
+    E.g. ``create_colormap([(0, 0, 0), (1, 1, 1)])`` creates the gray colormap.
+
+    If a dict is given, it must have at least one field of "r", "g", "b", "a".
+    Each value is a sequence of 2 values (i.e. an Nx2 array-like), where the
+    first value specifies the index (0..1), and the second the value for that color.
```

Comment:
This was there in visvis, enabling the colormap editor. I thought it might be handy.

<img src='https://raw.githubusercontent.com/wiki/almarklein/visvis/images/mip1.png'>